@xen-orchestra/fs/src/smb-mount.js

```diff
@@ -0,0 +1,35 @@
+import { join } from 'path'
+import { parse } from 'xo-remote-parser'
+import { tmpdir } from 'os'
+
+import MountHandler from './_mount'
+
+export default class SmbMountHandler extends MountHandler {
+  constructor(
+    remote,
+    { mountsDir = join(tmpdir(), 'xo-smb-mounts'), ...opts } = {}
+  ) {
+    const { domain, host, password, username } = parse(remote.url)
+    super(remote, opts, {
+      type: 'cifs',
+      host: `\\\\${host}`,
+      options: `user=${username},password=${password},domain=${domain}`,
+    })
+
+    if (this._remote.path === '') {
+      this._realPath = join(
+        mountsDir,
+        remote.id ||
+          Math.random()
+            .toString(36)
+            .slice(2)
+      )
+    } else {
+      this._realPath = this._remote.path
```

Comment:
Nope.

What is `_realPath`?
What is `remote.path`?

Reply:
`remote.path` is the place given by the user for mouting.

So I thought we were using `_realPath` to define the path to use. Either the user gives us a precise path, otherwise we use a `math.random` to define one by default.